source/slang/parser.cpp

```diff
@@ -598,10 +646,57 @@ namespace Slang
         AddModifier(&modifierLink, modifier);
     }
 
+    // 
+    // '::'? identifier ('::' identifier)* 
+    static Token parseAttributeName(Parser* parser)
+    {
+        const SourceLoc scopedIdSourceLoc = parser->tokenReader.PeekLoc();
+
+        // Strip initial :: if there is one
+        const TokenType initialTokenType = parser->tokenReader.PeekTokenType();
+        if (initialTokenType == TokenType::Scope)
```

Comment:
The `Parser` interface is kind of mess because it mixes member and non-member functions, so this might not have been obvious, but there is a function that captures the idiom of "if next token is X, then consume it". This logic could maybe be simplified to:

```c++
bool isScoped = AdvanceIf(parser, TokenType::Scope);
```